src/mlpack/methods/ann/activation_functions/selu_function.hpp

```diff
@@ -0,0 +1,164 @@
+/**
+ * @file selu_function.hpp
+ * @author Dakshit Agrawal
+ *
+ * Definition and implementation of the SELU function as introduced by
+ * Klambauer et. al. in Self Neural Networks.  The SELU activation
+ * function keeps the mean and variance of the input invariant.
+ *
+ * For more information, see the following paper.
+ *
+ * @code
+ * @article{Klambauer2017,
+ *   author  = {Gunter Klambauer and Thomas Unterthiner and
+ *              Andreas Mayr},
+ *   title   = {Self-Normalizing Neural Networks},
+ *   journal = {Advances in Neural Information Processing Systems},
+ *   year    = {2017}
+ * }
+ * }
+ * @endcode
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_SELU_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_SELU_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <algorithm>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The SELU activation function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *    lambda * x & : x > 0 \\
+ *    lambda * alpha(e^x - 1) & : x \le 0
+ *   \end{array}
+ * \right. \\
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     lambda & : x > 0 \\
+ *     lambda * (y + alpha) & : x \le 0
+ *   \end{array}
+ * \right.
+ * @f}
+ *
+ *
+ *
+ * NOTE:
+ * Make sure to use this activation function with normalized inputs and
+ * weights initialized with Lecun Normal Initialization.
+ */
+
+class SELUFunction
+{
+ public:
+    /**
+     * Computes the SELU activation function.
+     *
+     * @param x Input data.
+     * @return f(x).
+     */
+    static double Fn(const double x)
+    {
+        if (x < DBL_MAX) {
```

Comment:
Why is this check necessary?

Reply:
SELU does not have an upper bound, hence the need for a check.  The return value that I have set is a random value.